### PR TITLE
Feature/localizationsystem

### PR DESCRIPTION
**Pull request by CudlPudl**
Base files for the localization system, which "should work on paper". Will create an editor window / tool for adding localization entries and managing the language files later.

### CODE REVIEW

File: Assets/Scripts/Localization/LocalizationManager.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        public delegate void LanguageChanged(string language);
+        private LanguageChanged _onLanguageChanged;
+        private Dictionary<string, string> _localizations;
+        private string _previousLanguage;
+        public string CurrentLanguage { get; private set; }
+
+        public void Initialize()
+        {
+            if (PlayerPrefs.HasKey("CurrentLanguage"))
+                CurrentLanguage = PlayerPrefs.GetString("CurrentLanguage");
```
Comment (Taikatassu) [inline Assets/Scripts/Localization/LocalizationManager.cs]: Looks like you meant to use the `c_currentLanguageKey `const here?

File: Assets/Scripts/Localization/LocalizationManager.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            else
+                SetFirstLanguage();
+
+            LoadLocalization(CurrentLanguage);
+        }
+
+        private void SetFirstLanguage()
+        {
+            CurrentLanguage = "English";
+            PlayerPrefs.SetString("CurrentLanguage", CurrentLanguage);
```
Comment (Taikatassu) [inline Assets/Scripts/Localization/LocalizationManager.cs]: Same here: don't forget the `c_currentLanguageKey `const

File: Assets/Scripts/Localization/LocalizationManager.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                CurrentLanguage = PlayerPrefs.GetString("CurrentLanguage");
+            else
+                SetFirstLanguage();
+
+            LoadLocalization(CurrentLanguage);
+        }
+
+        private void SetFirstLanguage()
+        {
+            CurrentLanguage = "English";
```
Comment (Taikatassu) [inline Assets/Scripts/Localization/LocalizationManager.cs]: Consider promoting "English" to a const string.

File: Assets/Scripts/Localization/LocalizationManager.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            _previousLanguage = language;
+            _onLanguageChanged?.Invoke(language);
+        }
+
+        public void LoadLanguageInfo(string language)
+        {
+            var locAsset = Resources.Load<LocalizationAsset>($"Localization/LocalizationAsset_{language}.asset");
+            if (locAsset == null)
+            {
+                Debug.LogError("Localization error: failed to load localization file.");
```
Comment (Taikatassu) [inline Assets/Scripts/Localization/LocalizationManager.cs]: Consider including the name of the missing file in the error message.

File: Assets/Scripts/Localization/LocalizationKeys.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+using UnityEngine;
+
+namespace PotentialRobot.Localization
+{
+    [CreateAssetMenu(menuName = "Localization/Localization Keys", fileName = "LocalizationKeys.asset")]
+    public class LocalizationKeys : ScriptableObject
+    {
+        [field: SerializeField]
+        public List<string> Keys { get; private set; }
+    }
```
Comment (Taikatassu) [inline Assets/Scripts/Localization/LocalizationKeys.cs:12]: What is this asset used for? Why is this needed when the LocalizationAsset class exists?

Comment (CudlPudl) [reply Assets/Scripts/Localization/LocalizationKeys.cs:12]: Used later for the editor to cross reference all the language assets whether they have the required keys or not. Not really needed for runtime.

File: Assets/Scripts/Localization/LocalizationManager.cs (hunk 1 above)
Comment (CudlPudl) [reply Assets/Scripts/Localization/LocalizationManager.cs]: Changed to use the c_currentLanguageKey constant string

File: Assets/Scripts/Localization/LocalizationManager.cs (hunk 2 above)
Comment (CudlPudl) [reply Assets/Scripts/Localization/LocalizationManager.cs]: Changed to use the c_currentLanguageKey constant string

File: Assets/Scripts/Localization/LocalizationManager.cs (hunk 3 above)
Comment (CudlPudl) [reply Assets/Scripts/Localization/LocalizationManager.cs]: Promoted string to a const string.

File: Assets/Scripts/Localization/LocalizationManager.cs (hunk 4 above)
Comment (CudlPudl) [reply Assets/Scripts/Localization/LocalizationManager.cs]: Added full path of the file.

**Review by Taikatassu — APPROVED** [review]
